sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JacksonGenerator.scala

```diff
@@ -123,8 +122,9 @@ private[sql] class JacksonGenerator(
     case _ =>
       (row: SpecializedGetters, ordinal: Int) =>
         val v = row.get(ordinal, dataType)
-        sys.error(s"Failed to convert value $v (class of ${v.getClass}}) " +
-          s"with the type of $dataType to JSON.")
+        throw new SparkSQLJsonProcessingException(
+          s"Failed to convert value $v (class of ${v.getClass}}) " +
+            s"with the type of $dataType to JSON.")
```

Comment:
I would avoid this change since its throwing a private exception type to the user now.